client/displayservers/Wayland/gl.c

```diff
@@ -67,9 +68,61 @@ EGLDisplay waylandGetEGLDisplay(void)
   return eglGetDisplay(native);
 }
 
-void waylandEGLSwapBuffers(EGLDisplay display, EGLSurface surface)
+void waylandEGLSwapBuffers(EGLDisplay display, EGLSurface surface, const struct Rect * damage, int count)
 {
-  eglSwapBuffers(display, surface);
+  if (wlWm.eglSwapWithDamage == SWAP_WITH_DAMAGE_UNKNOWN)
+  {
+    const char *exts = eglQueryString(display, EGL_EXTENSIONS);
+    if (wl_proxy_get_version((struct wl_proxy *) wlWm.surface) < 4)
+    {
+      wlWm.eglSwapWithDamage = SWAP_WITH_DAMAGE_UNSUPPORTED;
+      DEBUG_INFO("Swapping buffers with damage: not supported, need wl_compositor v4");
+    }
+    else if (strstr(exts, "EGL_KHR_swap_buffers_with_damage") && g_egl_dynProcs.eglSwapBuffersWithDamageKHR)
+    {
+      wlWm.eglSwapWithDamage = SWAP_WITH_DAMAGE_KHR;
```

Comment:
The prototypes of KHR and EXT look identical (unless I missed something), why not store the function pointer at walyandInit to avoid these conditionals running on every call?

Reply:
We are unable to query the EGL extensions without a context, which is created in the `EGL` renderer. The only place we are guaranteed to have an EGL context active is during the `waylandEGLSwapBuffers` call.

Reply:
Ok, but even still there is an additional check on the enum below to decide which method to call, why not store the pointer to the correct one and just call it?

Reply:
Good idea, changed.